groclient/lib.py

```diff
@@ -403,20 +407,35 @@ def get_source_ranking(access_token, api_host, series):
     return get_data(url, headers, params).json()
 
 
-def rank_series_by_source(access_token, api_host, series_list):
-    for series in series_list:
+def rank_series_by_source(access_token, api_host, selections_list):
+    series_map = OrderedDict()
+    for selection in selections_list:
+        series_key = '.'.join([str(selection.get(type_id))
+                               for type_id in DATA_SERIES_UNIQUE_TYPES_ID
+                               if type_id != 'source_id'])
+        if series_key not in series_map:
+            series_map[series_key] = {}
+        elif None in series_map[series_key]:
+            continue
+        series_map[series_key][selection.get('source_id')] = selection
```

Comment:
This is minor, we could capture the string 'source_id' in a variable as it occurs several times in the function

Reply:
Because of the strong backwards compatibility guarantees the API has, there's no chance of that key changing in the future. So, I don't think a variable would be very different from the constant value. If anything, maybe a little greater cognitive complexity since then you have another variable to mentally decode when reading the code